python/dgl/backend/pytorch/tensor.py

```diff
@@ -81,7 +81,8 @@ def copy_to(input, ctx):
     if ctx.type == 'cpu':
         return input.cpu()
     elif ctx.type == 'cuda':
-        th.cuda.set_device(ctx.index)
+        if ctx.index is not None:
```

Comment:
Is none allowed for the index field?

Reply:
Yes when it is initiated as torch.device('cuda') instead of torch.device('cuda:0')